src/JestProcessManagement/JestProcess.ts

```diff
@@ -88,7 +88,7 @@ export class JestProcess {
     })
   }
 
-  public runJestWithUpdateForSnapshots(callback) {
+  public runJestWithUpdateForSnapshots(callback: Function) {
```

Comment:
can we be more explicit about the function signature?

Reply:
`JestExt` uses it with an empty-parameter callback, and `Runner` is even looser with a `runJestWithUpdateForSnapshots(completion: any): void` signature. Internally, it calls it as `completion();`. So if you have advice on a better signature I'm all ears (I'm relatively new to Typescript).

Reply:
in `jest-editor-support`:
```
runJestWithUpdateForSnapshots(completion: any, args: string[]) 
```
indeed the completion signature is too loose, it should have been `() => void`, and the `args` actually is optional, I will fix it on the jest-editor-support side. As for this PR, I think we should do the right thing by declaring  
```
public runJestWithUpdateForSnapshots(callback: () => void) 
```

Note: a bit background for changing 'Function': tslint recommendation [ban-types](https://github.com/palantir/tslint/blob/3efdfe57b7c0eb1b21e21cec128c993cf20da438/src/configs/recommended.ts#L28)

Reply:
Hm, I installed the TSLint extension I was missing, but the project doesn't seem configured to hint for that rule at the moment. Is there something I'm missing, or is this just advice based on your previous experience with that rule?

Reply:
we probably should have added tslint sooner, but at least it will be there after #447, which I have already fixed a lot of `Function` types... 😉